eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/NoImplInPublicAPI.java

```diff
@@ -50,7 +50,7 @@ public void visitToken(DetailAST ast) {
         if (ast.getType() == TokenTypes.PACKAGE_DEF) {
             String packageName = FullIdent.createFullIdent(ast.findFirstToken(TokenTypes.DOT)).getText();
             this.isTrackTwo = packageName.startsWith(COM_AZURE);
-            this.isImplPackage = packageName.startsWith(COM_AZURE_CORE_IMPLEMENTATION);
+            this.isImplPackage = packageName.endsWith(DOT_IMPLEMENTATION);
```

Comment:
This is still not right! It might be that we have packages under the implementation package, e.g. `com.azure.implementation.logging`

Reply:
updated with a regular expression pattern match: 
private static final String IMPL_PACKAGE_REGULAR_EXPR = 
"^com\\.azure(\\..*)+\\.implementation(\\..*)*";